x-pack/plugins/cases/server/routes/api/files/post_file.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { CASE_FILES_URL, MAX_FILE_SIZE } from '../../../../common/constants';
+import { createCaseError } from '../../../common/error';
+import { createCasesRoute } from '../create_cases_route';
+import type { attachmentApiV1 } from '../../../../common/types/api';
+import type { caseDomainV1 } from '../../../../common/types/domain';
+
+export const postFileRoute = createCasesRoute({
+  method: 'post',
+  path: CASE_FILES_URL,
+  params: {
+    params: schema.object({
+      case_id: schema.string(),
+    }),
+  },
+  routerOptions: {
+    access: 'public',
+    summary: 'Attach a file to a case',
+    tags: ['oas-tag:cases'],
+    body: {
+      /*
+       * The maximum file size allowed here is the upper limit allowed when registering the cases file
+       * kinds. We register the file kinds in x-pack/plugins/cases/public/files/index.ts.
+       *
+       * If the user configured a custom value the validation will still fail when calling the service.
+       */
+      maxBytes: MAX_FILE_SIZE,
+      accepts: 'multipart/form-data',
+    },
+  },
+  handler: async ({ context, request, response }) => {
+    try {
+      const caseContext = await context.cases;
+      const casesClient = await caseContext.getCasesClient();
+
+      const fileRequest = request.body as attachmentApiV1.PostFileAttachmentRequest;
+      const caseId = request.params.case_id;
+
+      const res: caseDomainV1.Case = await casesClient.attachments.addFile({ fileRequest, caseId });
```

Comment:
I think, in the cases client (`addFile` method), we change a bit the signature of the function to be like this:

```
{
 file: Readable;
 fileName: string;
 mimeType: string;
 caseId: string;
$abort?: ReplaySubject<unknown>;
}
```

This way it is clearer to the cases client consumer that we expect a `Readable` (and not `unknown`) type for the file and we operate with streams from now on.	

Lastly, about the `$abort` we can do

```
// On the route

const abort$ = new ReplaySubject();
const sub = req.events.aborted$.subscribe(abort$);

try {addFile({ file, fileName, caseId, $abort }) } catch { ... } finally {
    sub.unsubscribe();
  }
```

Reply:
My original intention was to have `file: Readable`, but `io-ts` doesn't have a `Readable` schema.

Maybe I can separate `PostFileAttachmentRequest` from `PostFileAttachmentRequestRt` and have it as `unknown` in the schema.

Then we have a clear type for AddFile and a schema that we use internally for decoding the params.

Reply:
I just saw `x-pack/plugins/cases/server/client/attachments/types.ts`. LGTM!